### PR TITLE
Add GraphQL client BBE

### PR DESCRIPTION
**Pull request by MohamedSabthar**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: examples/graphql-client/graphql_client.bal [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+type Person record {
+    string name;
+    int age;
+};
+
+public function main() returns error? {
+    // Creates a new client with the backend URL.   
+    graphql:Client graphqlClient = check new ("http://localhost:4000/graphql");
+
+    string document = "{ profile { name, age } }";
```
Comment (ThisaruGuruge) [inline examples/graphql-client/graphql_client.bal:28]:
```suggestion
    string document = "{ profile { name, age } }";

```

File: examples/graphql-client/graphql_client.bal [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    // The `execute()` remote function of graphql:Client takes a GraphQL document 
+    // as the required argument and sends a request to the specified backend URL
+    // seeking a response. On the retrieval of a successful response, the client
+    // tries to perform data binding for the user-defined data type. On failure to
+    // retrieve a successful response or when the client fails to perform data
+    // binding, a graphql:ClientError will be returned.
+
```
Comment (ThisaruGuruge) [inline examples/graphql-client/graphql_client.bal:35]:
```suggestion
```

File: examples/graphql-client/graphql_client.bal [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    ResponseWithErrors response = check graphqlClient->execute(document);
```
Comment (ThisaruGuruge) [inline examples/graphql-client/graphql_client.bal:36]: Shall we add an error handling case here to make it clear?

Comment (MohamedSabthar) [reply examples/graphql-client/graphql_client.bal:36]: To make the bbe simple error handling is omitted here. 
Similar pattern followed in other client examples (http, udp, tcp etc)

Comment (ThisaruGuruge) [reply examples/graphql-client/graphql_client.bal:36]: Agreed.

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**